src/web/crate_details.rs

```diff
@@ -46,72 +46,61 @@ pub struct CrateDetails {
-        m.insert(
-            "last_successful_build".to_string(),
-            self.last_successful_build.to_json(),
-        );
-        m.insert("rustdoc_status".to_string(), self.rustdoc_status.to_json());
-        m.insert("repository_url".to_string(), self.repository_url.to_json());
-        m.insert("homepage_url".to_string(), self.homepage_url.to_json());
-        m.insert("keywords".to_string(), self.keywords.to_json());
-        m.insert("have_examples".to_string(), self.have_examples.to_json());
-        m.insert("target_name".to_string(), self.target_name.to_json());
-        m.insert("releases".to_string(), self.releases.to_json());
-        m.insert("github".to_string(), self.github.to_json());
-        m.insert("github_stars".to_string(), self.github_stars.to_json());
-        m.insert("github_forks".to_string(), self.github_forks.to_json());
-        m.insert("github_issues".to_string(), self.github_issues.to_json());
-        m.insert("metadata".to_string(), self.metadata.to_json());
-        m.insert("is_library".to_string(), self.is_library.to_json());
-        m.insert("yanked".to_string(), self.yanked.to_json());
-        m.insert("doc_targets".to_string(), self.doc_targets.to_json());
-        m.insert("license".to_string(), self.license.to_json());
-        m.insert(
-            "documentation_url".to_string(),
-            self.documentation_url.to_json(),
-        );
-        m.to_json()
+
+        state.serialize_field("release_time", &duration_to_str(self.release_time))?;
+        state.serialize_field("build_status", &self.build_status)?;
+        state.serialize_field("last_successful_build", &self.last_successful_build)?;
+        state.serialize_field("rustdoc_status", &self.rustdoc_status)?;
+        state.serialize_field("repository_url", &self.repository_url)?;
+        state.serialize_field("homepage_url", &self.homepage_url)?;
+        state.serialize_field("keywords", &self.keywords)?;
+        state.serialize_field("have_examples", &self.have_examples)?;
+        state.serialize_field("target_name", &self.target_name)?;
+        state.serialize_field("releases", &self.releases)?;
+        state.serialize_field("github", &self.github)?;
+        state.serialize_field("github_stars", &self.github_stars)?;
+        state.serialize_field("github_forks", &self.github_forks)?;
+        state.serialize_field("github_issues", &self.github_issues)?;
+        state.serialize_field("metadata", &self.metadata)?;
+        state.serialize_field("is_library", &self.is_library)?;
+        state.serialize_field("doc_targets", &self.doc_targets)?;
+        state.serialize_field("yanked", &self.yanked)?;
+        state.serialize_field("license", &self.license)?;
+        state.serialize_field("documentation_url", &self.documentation_url)?;
+
+        state.end()
     }
 }
 
-#[derive(Debug, Eq, PartialEq)]
+#[derive(Debug, Eq, PartialEq, serde::Serialize)]
```

Comment:
nit: can you use `Serialize` instead? I think it looks nicer and serde is pretty clear from context.